File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cs
public class Solution {
    public IList<int> FindDisappearedNumbers(int[] nums) {
        IList<int> ans = new List<int>();

        foreach(int num in nums){
            int indx = Math.Abs(num)-1;
            if(nums[indx] > 0){
            nums[indx] *= -1;
            }
        }
        for(int i=0;i<nums.Length;i++){
            if(nums[i] > 0){
                ans.Add(i+1);
            }
        }
        return ans;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> ans;
        int n = nums.size();
        vector<int> freq(n+1,0);

        for(int n : nums){
            freq[n]++;
        }

        for(int i=1;i<=n;i++){
            if(!freq[i]){
                ans.push_back(i);
            }
        }
        return ans;
    }
};